### PR TITLE
Rb/fixed branch

### PR DESCRIPTION
**Pull request by RohitBhosale204**

### CODE REVIEW

File: src/core/aws/aws-project.ts [hunk 1]
```diff
@@ -313,14 +316,12 @@ export default class AWSProject extends BaseProject {
...
                 command += ` -var-file=${varFile}`;
             }
-            execSync(command, {
-                cwd: `${projectPath}`,
-                stdio: 'inherit',
-                env: process.env
-            });
-            AppLogger.info('Terraform destroy completed successfully.', true);
+    
+            const result = executeCommandWithRetry(command, {cwd: `${projectPath}`, stdio: 'inherit', env: "process.env" })
```
Comment (salim-jujara) [inline src/core/aws/aws-project.ts:320]: what will be value of result? Also, we are not checking or doing anything with result..

File: src/core/common-functions/execCommands.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    env?:ShellOption;
+}
+
+export async function executeCommandWithRetry(
+    command: string,
+    options: ExecuteCommandOptions = {} 
+) {
+    const {
+        cwd = process.cwd(),
+        maxRetries = 3,
```
Comment (salim-jujara) [inline src/core/common-functions/execCommands.ts:21]: Is it always going to be 3 retries? Can this be passed as parameter? Also, make it configurable based on system/project settings

File: src/core/common-functions/execCommands.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        shell = undefined
+    } = options;
+
+    let attempts = 0;
+    let success = false;
+
+    while (attempts < maxRetries && !success) {
+        try {
+            attempts++;
+            execSync(command, { cwd, stdio, shell, env });
```
Comment (salim-jujara) [inline src/core/common-functions/execCommands.ts:32]: Again, here.. we are not checking what will be returned by execSync. We do have to capture it and also take some action based on it..

File: src/core/setup-application.ts [hunk 4]
```diff
@@ -409,7 +297,7 @@ export default class CreateApplication extends BaseProject {
                 if (url) {
                     AppLogger.debug(`Deleting repository for..., ${url}`);
                     const command = `curl -X DELETE -u "${userName}:${token}" ${url}`;
-                    const result = execSync(command, { stdio: 'pipe' });
+                    const result = executeCommandWithRetry(command, { stdio: 'pipe' })
                     AppLogger.debug(`Repository deleted successfully:, ${result.toString()}`);
```
Comment (salim-jujara) [inline src/core/setup-application.ts:301]: what is the value of result? also, without checking it how is it considered to be deleted successfully...

File: src/core/aws/aws-project.ts (hunk 1 above)
Comment (RohitBhosale204) [reply src/core/aws/aws-project.ts:320]: YEs, we are taking that out

File: src/core/common-functions/execCommands.ts (hunk 2 above)
Comment (RohitBhosale204) [reply src/core/common-functions/execCommands.ts:21]: Yes, Now we are passing the maxretries as a parameter.

File: src/core/setup-application.ts (hunk 4 above)
Comment (RohitBhosale204) [reply src/core/setup-application.ts:301]: I have added the try catch block over it. Now we are checking if there is any error during deleting the repositories.

**Review by VishalBikkad — APPROVED** [review]

**Review by anurag-calfus — APPROVED** [review]

**Pull request merged**